ingestion-beam/src/main/java/com/mozilla/telemetry/decoder/MessageScrubber.java

```diff
@@ -6,28 +6,36 @@
 import com.google.common.annotations.VisibleForTesting;
 import com.google.common.collect.Streams;
 import com.mozilla.telemetry.ingestion.core.Constant.Attribute;
+import com.mozilla.telemetry.transforms.MapElementsWithErrors.MessageShouldBeDroppedException;
 import java.util.Map;
 import java.util.Optional;
 import java.util.stream.Stream;
-import org.apache.beam.sdk.metrics.Counter;
 import org.apache.beam.sdk.metrics.Metrics;
 
 public class MessageScrubber {
 
-  private static final Counter countScrubbedBug1567596 = Metrics.counter(MessageScrubber.class,
-      "bug_1567596");
-  private static final Counter countScrubbedBug1562011 = Metrics.counter(MessageScrubber.class,
-      "bug_1562011");
-  private static final Counter countRedactedBug1602844 = Metrics.counter(MessageScrubber.class,
-      "bug_1602844");
+  private enum ScrubAction {
+    DROP, REDACT, SEND_TO_ERRORS
+  }
+
+  /**
+   * Special exception to signal that a message is affected by a specific bug and should
+   * be written to error output.
+   */
+  public static class AffectedByBugException extends Exception {
+
+    public AffectedByBugException(String bugNumber) {
```

Comment:
We never do anything with the bugNumber here. In the constructor body here, we should call `super(...)` in order to pass a message like `"Affected by bug " + bugNumber`